src/storage-client/src/controller.rs

```diff
@@ -873,10 +874,22 @@ impl fmt::Display for StorageError {
                     id.iter().map(|id| id.to_string()).join(", ")
                 )
             }
-            Self::ClientError(err) => write!(f, "underlying client error: {:#}", err),
-            Self::IOError(err) => write!(f, "failed to read or write state: {err}"),
-            Self::DataflowError(err) => write!(f, "dataflow failed to process request: {err}"),
-            Self::InvalidUsage(err) => write!(f, "invalid usage: {err}"),
+            Self::ClientError(err) => {
+                // WIP: Should we even try and print the causes here? Feels like thats the
+                // responsibility of the caller now.
```

Comment:
Oh, right! So `fmt::Display` implementations now should _never_ try to print causes, right? And instead they should be very diligent about reporting `source()`s correctly, right?

Reply:
I _think_ so, but I have to double check, that's why I left in this WIP/TODO comment.